src/main/protobuf/soc/inventory.proto

```diff
@@ -0,0 +1,70 @@
+syntax = "proto2";
+
+package protos.soc;
+
+enum Resource {
+    BRICK = 0;
+    ORE = 1;
+    SHEEP = 2;
+    WHEAT = 3;
+    WOOD = 4;
+}
+
+enum DevelopmentCard {
+    KNIGHT = 0;
+    YEAR_OF_PLENTY = 1;
+    MONOPOLY = 2;
+    ROAD_BUILDER = 3;
+    POINT = 4;
+}
+
+message ResourceCount {
+    required Resource cardType = 1;
+    required int32 count = 2;
+}
+
+message DevelopmentCardSet {
+    map<int32, DevelopmentCard> development_cards = 1;
+}
+
+message ProbableResourceSet {
+    message ProbableCardValue {
+        required Resource res_type = 1;
+        required int32 known_amount = 2;
+        required double unknown_amount = 3;
+    }
+    repeated ProbableCardValue probable_resource_cards = 1;
+}
+
+message PossibleHands {
+
+    message HandCombination {
+        message HandsWithMultiplier {
+            repeated ResourceCount cards = 1;
+            required int32 multiplier = 2;
+        }
+        map<int32, HandsWithMultiplier> hand = 1;
+    }
+    repeated HandCombination hands = 1;
+}
+
+message PublicInventory {
+    required int32 num_cards = 1;
+    required int32 num_development_cards = 2;
+    map<int32, DevelopmentCard> played_dev_cards = 3;
+}
+
+message PerfectInventory {
```

Comment:
[minor] `PrivateInventory`

Reply:
I would recommend having a type `DevelopmentCardSpecification`

```
message DevelopmentCardSpecification {
     required DevelopmentCard type = 1;
     required isPlayed = 2;
     optional turnPurchased = 3;
     optional turnPlayed = 4;
}

Then instead of having a `DevelopmentCardSet`, you can just have a `repeated DevelopmentCardSpecification`

It's parallel with `repeated ResourceCount`.